Tools/LyTestTools/ly_test_tools/o3de/material_canvas_test.py

```diff
@@ -0,0 +1,174 @@
+        :param instance: Parent MaterialCanvasTestSuite.MultiTestCollector instance executing the test
+        :param request: PyTest request object
+        :param workspace: LyTestTools workspace manager
+        """
+        pass
+
+    @staticmethod
+    def wrap_run(instance: MaterialCanvasTestSuite.MultiTestCollector,
+                 request: _pytest.fixtures.FixtureRequest,
+                 workspace: AbstractWorkspaceManager,
+                 material_canvas_test_results: MultiTestSuite.TestData) -> None:
+        """
+        User-overrideable wrapper function, which will run both before and after test.
+        Any code before the 'yield' statement will run before the test. With code after yield run after the test.
+        Setup will run before wrap_run starts. Teardown will run after it completes.
+        :param instance: Parent MaterialCanvasTestSuite.MultiTestCollector instance executing the test
+        :param request: PyTest request object
+        :param workspace: LyTestTools workspace manager
+        :param material_canvas_test_results: Currently recorded MaterialCanvas test results
+        """
+        yield
+
+    @staticmethod
+    def teardown(instance: MaterialCanvasTestSuite.MultiTestCollector,
+                 request: _pytest.fixtures.FixtureRequest,
+                 workspace: AbstractWorkspaceManager,
+                 material_canvas_test_results: MultiTestSuite.TestData) -> None:
+        """
+        User-overrideable teardown function, which will run after the test
+        :param instance: Parent MaterialCanvasTestSuite.MultiTestCollector instance executing the test
+        :param request: PyTest request object
+        :param workspace: LyTestTools workspace manager
+        :param material_canvas_test_results: Currently recorded MaterialCanvas test results
+        """
+        pass
+
+
+class MaterialCanvasSharedTest(SharedTest):
+    """
+    Test that will run in parallel with tests in different MaterialCanvas instances, as well as serially batching
+    with other tests in each MaterialCanvas instance. Minimizes total test run duration.
+
+    Does not support per test setup/teardown to avoid creating race conditions
+    """
+    # Specifies if the test can be batched in the same MaterialCanvas
+    is_batchable = True
+    # Specifies if the test can be run in multiple MaterialCanvases in parallel
+    is_parallelizable = True
+
+
+class MaterialCanvasParallelTest(MaterialCanvasSharedTest):
+    """
+    Test that will run in parallel with tests in different MaterialCanvas instances,
+    though not serially batched with other tests in each MaterialCanvas instance.
+    Reduces total test run duration, while limiting side-effects between tests.
+
+    Does not support per test setup/teardown to avoid creating race conditions
+    """
+    is_batchable = False
+    is_parallelizable = True
+
+
+class MaterialCanvasBatchedTest(MaterialCanvasSharedTest):
```

Comment:
same not clear to me why this implements MaterialCanvasSharedTest while MaterialCanvasSharedTest(SharedTest). this technically works, but it seems odd that only MaterialCanvasSharedTest implements from a base class while these implement from MaterialCanvasSharedTest.